### PR TITLE
Agent description blog post

### PR DESCRIPTION
**Pull request by afourney**
## Why are these changes needed?
Adds a blog post describing the correct use of the description field on 0.2.2

## Related issue number

#736 

## Checks

- [ ] I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [ ] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [ ] I've made sure all auto checks have passed.

### CODE REVIEW

File: website/blog/2023-12-29-AgentDescriptions/index.mdx [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+{self._participant_roles(agents)}.
+
+Read the following conversation.
+Then select the next role from {[agent.name for agent in agents]} to play. Only return the role.
+```
+
+Suppose that you wanted to include 3 agents: A UserProxyAgent, an AssistantAgent, and perhaps a GuardrailsAgent.
+
+Prior to 2.2, this template would expand to:
```
Comment (ekzhu) [inline website/blog/2023-12-29-AgentDescriptions/index.mdx:42]: 0.2.2

File: website/blog/2023-12-29-AgentDescriptions/index.mdx [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+title: "All About Agent Descriptions"
+authors:
+  - afourney
+tags: [AutoGen]
+---
+
+
+## TLDR
+
+AutoGen 0.2.2 introduces a [description](https://microsoft.github.io/autogen/docs/reference/agentchat/conversable_agent#__init__) field to ConversableAgent (and all subclasses), and changes GroupChat so that it uses agent `description`s rather than `system_prompt`s when choosing which agents should speak next.
```
Comment (sonichi) [inline website/blog/2023-12-29-AgentDescriptions/index.mdx:11]: system_prompt -> system_message
to match the code

File: website/blog/2023-12-29-AgentDescriptions/index.mdx [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+Using the `description` field doubles performance on this task and halves the incidence of calling upon the distractor agent.
+
+## Tips for Writing Good Descriptions
+Since `descriptions` serve a different purpose than `system_prompts`, it is worth reviewing what makes a good agent description. While descriptions are new, the following tips appear to lead to good results:
+
+- Avoid using the 1st or 2nd person perspective. Descriptions should not contain "I" or "You", unless perhaps "You" is in reference to the GroupChat / orchestrator
+- Include any details that might help the orchestrator know when to call upon the agent
+- Keep descriptions short (e.g., "A helpfupl AI assistant with strong natural language and Python coding skills.").
+
+The main thing to remember is that the description is for the benefit of the GroupChatManager, not for the Agent's own use or instruction.
```
Comment (sonichi) [inline website/blog/2023-12-29-AgentDescriptions/index.mdx:135]:
```suggestion
The main thing to remember is that **the description is for the benefit of the GroupChatManager, not for the Agent's own use or instruction.**
```

**Review by ekzhu — APPROVED** [review]

File: website/blog/2023-12-29-AgentDescriptions/index.mdx (hunk 1 above)
Comment (afourney) [reply website/blog/2023-12-29-AgentDescriptions/index.mdx]: Thanks. Fixed.

File: website/blog/2023-12-29-AgentDescriptions/index.mdx (hunk 2 above)
Comment (afourney) [reply website/blog/2023-12-29-AgentDescriptions/index.mdx]: Thanks. Fixed here, and throughout.

File: website/blog/2023-12-29-AgentDescriptions/index.mdx (hunk 3 above)
Comment (afourney) [reply website/blog/2023-12-29-AgentDescriptions/index.mdx]: Bolded. Thanks.

**Review by LittleLittleCloud — APPROVED** [review]
LGTM, thanks

**Review by JieyuZ2 — APPROVED** [review]

**Review by sonichi — COMMENTED** [review]
Would you like to add an image in the beginning? For example, the plot of the performance change.

**Review by LinxinS97 — APPROVED** [review]
LGTM!

**Review by ekzhu — APPROVED** [review]

**Pull request merged**